src/pydash/objects.py

```diff
@@ -1320,6 +1324,119 @@ def deep_iteratee(value, key):
         return callit(iteratee, obj, properties)
 
 
+def apply(obj: T, func: t.Callable[[T], T2]) -> T2:
+    """
+    Returns the result of calling `func` on `obj`. Particularly useful to pass
+    `obj` through a function during a method chain.
+
+    Args:
+        obj: Object to apply function to
+        func: Function called with `obj`.
+
+    Returns:
+        Results of ``func(value)``.
+
+    Example:
+
+        >>> apply(5, lambda x: x * 2)
+        10
+
+    .. versionadded:: 8.0.0
+    """
+    return func(obj)
+
+
+def apply_if(obj: T, func: t.Callable[[T], T2], predicate: t.Callable[[T], bool]) -> t.Union[T, T2]:
+    """
+    Apply `func` to `obj` if `predicate` returns `True`.
+
+    Args:
+        obj: Object to apply `func` to.
+        func: Function to apply to `obj`.
+        predicate: Predicate applied to `obj`.
+
+    Returns:
+        Result of applying `func` to `obj` or `obj`.
+
+    Example:
+
+        >>> apply_if(2, lambda x: x * 2, lambda x: x > 1)
+        4
+        >>> apply_if(2, lambda x: x * 2, lambda x: x < 1)
+        2
+
+    .. versionadded:: 8.0.0
+    """
+    return func(obj) if predicate(obj) else obj
+
+
+def apply_if_not_none(obj: t.Optional[T], func: t.Callable[[T], T2]) -> t.Optional[T2]:
+    """
+    Apply `func` to `obj` if `obj` is not ``None``.
+
+    Args:
+        obj: Object to apply `func` to.
+        func: Function to apply to `obj`.
+
+    Returns:
+        Result of applying `func` to `obj` or ``None``.
+
+    Example:
+
+        >>> apply_if_not_none(2, lambda x: x * 2)
+        4
+        >>> apply_if_not_none(None, lambda x: x * 2) is None
+        True
+
+    .. versionadded:: 8.0.0
+    """
+    return apply_if(obj, func, lambda x: x is not None)  # type: ignore
+
+
+@t.overload
+def apply_ignore_excs(
+    obj: T, func: t.Callable[[T], T2], excs: t.Iterable[t.Type[Exception]], fallback: T3
+) -> t.Union[T2, T3]:
+    ...
+
+
+@t.overload
+def apply_ignore_excs(
+    obj: T, func: t.Callable[[T], T2], excs: t.Iterable[t.Type[Exception]], fallback: None = None
+) -> t.Union[T2, None]:
+    ...
+
+
+def apply_ignore_excs(obj, func, excs, fallback=None):
```

Comment:
What about naming this `apply_safely()` or `apply_with_catch()`?

For argument naming:

- `excs -> exceptions`: Prefer the full name here (we're using `exceptions` in another function as well)
- `fallback -> default`: `default` is used elsewhere to signify the value to return when can't return an actual value

Reply:
Thinking some more on this function. Wondering if we should have a way to specify that obj should be returned on exception without having to pass it in explicitly. For example, if using this in chaining, then the value of obj won't necessarily be known when the method is called so won't be able to pass it as the default/fallback.

Reply:
We could have the default value be UNSET that would return obj on exception.

Reply:
very good points, I applied everything.

About the naming I named it `apply_catch`. I find `apply_safely` a  bit misleading as it only catches specific exceptions and `apply_with_catch` sounds a bit funny to me. Let me know what you think.

Reply:
Sounds good to me!